test/Test/DecisionSpec.hs

```diff
@@ -25,3 +29,59 @@ test_decisionMonoidMempty = pure $ testProperty "Decision Monoid:Mempty" $ prope
 
 genDecision :: MonadGen m => m Decision
 genDecision = Gen.enumBounded
+
+genTextArr :: MonadGen m => m [Text]
+genTextArr = Gen.list (Range.constant 0 10) $ Gen.text (Range.constant 0 10) Gen.alpha
+
+genText :: MonadGen m => m Text
+genText = Gen.text (Range.constant 0 10) Gen.alpha
+
+genGhcVerArr :: MonadGen m => m [GhcVer]
+genGhcVerArr = Gen.list (Range.constant 0 10) Gen.enumBounded
+
+genCustomPrelude :: MonadGen m => m CustomPrelude
+genCustomPrelude = Prelude
+    <$> Gen.text (Range.constant 0 10) Gen.alpha
+    <*> Gen.text (Range.constant 0 10) Gen.alpha
+
+genLicense :: MonadGen m => m License
+genLicense = Gen.element licenseNames
+
+randomConfig :: MonadGen m => m PartialConfig
+randomConfig = do
+    text <- liftGen genText
+    list <- liftGen genTextArr
+    dec  <- liftGen genDecision
+    lic  <- liftGen genLicense
+    ghc  <- liftGen genGhcVerArr
+    lude <- liftGen genCustomPrelude
+    pure Config
+        { cOwner      = Last $ Just text
+        , cFullName   = Last $ Just text
+        , cEmail      = Last $ Just text
+        , cLicense    = Last $ Just lic
+        , cGhcVer     = Last $ Just ghc
+        , cCabal      = dec
+        , cStack      = dec
+        , cGitHub     = dec
+        , cTravis     = dec
```

Comment:
This is not a real test, because all fields will have the same value. I think it's better if you use `RecordWildCards` and write code like this:

```haskell
do
    cOwner <- Last . Just <$> genText
    cCabal <- genDecision
    ...
    pure Config{..}
```

Reply:
Will values be generated newly every time in that `<-`  case?